From 6b99e0cf9d2d4c76d1fcee9df2b13e94bbf7316d Mon Sep 17 00:00:00 2001
Message-Id: <6b99e0cf9d2d4c76d1fcee9df2b13e94bbf7316d.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Fri, 26 Dec 2014 11:01:45 +0530
Subject: [PATCH 19/28] MUST_REBASE [VPG]: video/adf/intel: check sprite plane
 visibility

When panel fitter is enabled, use the scaled source size to
calculate the plane visibility.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-4647
Change-Id: I9c6cff5a6c24285d50a75d644ff7b9bb2c90b103
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Signed-off-by: Zhang, Guangli <guangli.zhang@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_sp_plane.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
index 8e6ef69..30fca10 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
@@ -485,8 +485,15 @@ static int vlv_sp_validate(struct intel_plane *planeptr,
 
 	clip.x1 = 0;
 	clip.y1 = 0;
-	clip.x2 = mode.hdisplay;
-	clip.y2 = mode.vdisplay;
+	if (intel_pipe->pipe_reg.pfit_control & PFIT_ENABLE) {
+		clip.x2 = (((intel_pipe->pipe_reg.scaling_src_size >> 16) &
+					0x0000FFFF) + 1);
+		clip.y2 = (((intel_pipe->pipe_reg.scaling_src_size) &
+					0x0000FFFF) + 1);
+	} else {
+		clip.x2 = mode.hdisplay;
+		clip.y2 = mode.vdisplay;
+	}
 
 	if (get_format_config(buf->format, &format_config, &bpp,
 				config->alpha)) {
-- 
1.7.9.5

